programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -132,6 +132,101 @@ pub enum SyscallError {
 
 type Error = Box<dyn std::error::Error>;
 
+pub trait HasherImpl {
+    const HASH_BYTES: usize;
+    const NAME: &'static str;
+
+    fn create_hasher() -> Self;
+    fn hash(&mut self, val: &[u8]);
+    fn result(self) -> [u8; HASH_BYTES];
+    fn get_compute_budget() -> ComputeBudget;
+}
+
+pub struct Sha256Hasher(Hasher);
+pub struct Blake3Hasher(blake3::Hasher);
+pub struct Keccak256Hasher(keccak::Hasher);
+
+impl HasherImpl for Sha256Hasher {
+    const HASH_BYTES: usize = HASH_BYTES;
+    const NAME: &'static str = "Sha256";
+
+    fn create_hasher() -> Self {
+        Sha256Hasher(Hasher::default())
+    }
+
+    fn hash(&mut self, val: &[u8]) {
+        self.0.hash(val);
+    }
+
+    fn result(self) -> [u8; HASH_BYTES] {
+        self.0.result().to_bytes()
+    }
+
+    fn get_compute_budget() -> ComputeBudget {
+        let mut compute_budget: ComputeBudget = ComputeBudget::default();
+        compute_budget.hash_base_cost = 85;
```

Comment:
instead of abusing the compute budget here, let's add a method for each arg that _takes_ a `&ComputeBudget` and return the corresponding value
```
fn get_base_cost(compute_budget: &ComputeBudget) -> u64;
fn get_byte_cost(compute_budget: &ComputeBudget) -> u64;
fn get_max_slices() -> u64;
```

Reply:
added these above methods and they return the corresponding value, but as mentioned in the above conversation all hashing methods use the same `base_cost`, `max_salice`, and `byte_cost` since they never had different values so they should be taken care of in another PR/issue.